File: test_resources/rust/struct_with_parameters_buffers.rs
impl IntoVMBuffers for Test {
    fn read_from_buffers(bytes_reader: &mut BytesReader) -> Self {
        Self {
            delta_time: bytes_reader.read_f32(),
            view_width: bytes_reader.read_f32(),
            view_height: bytes_reader.read_f32(),
            touch_start_x: bytes_reader.read_f32(),
            touch_start_y: bytes_reader.read_f32(),
            last_touch_x: bytes_reader.read_f32(),
            last_touch_y: bytes_reader.read_f32(),
            touch_x: bytes_reader.read_f32(),
            touch_y: bytes_reader.read_f32(),
        }
    }

    fn write_to_buffers(&self, bytes_writer: &mut BytesWriter) {
        bytes_writer.write_f32(self.delta_time);
        bytes_writer.write_f32(self.view_width);
        bytes_writer.write_f32(self.view_height);
        bytes_writer.write_f32(self.touch_start_x);
        bytes_writer.write_f32(self.touch_start_y);
        bytes_writer.write_f32(self.last_touch_x);
        bytes_writer.write_f32(self.last_touch_y);
        bytes_writer.write_f32(self.touch_x);
        bytes_writer.write_f32(self.touch_y);
    }

    fn skip(bytes_reader: &mut BytesReader, count: u64) {
        for _ in 0..count {
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
            bytes_reader.read_f32();
        }
    }
}

File: test_resources/dart/struct_with_parameters_buffers.dart
class TestEmplaceToBuffers implements EmplaceToBuffers<Test> {
  const TestEmplaceToBuffers()

  @override
  void read(BytesReader reader, Test model) {
    model.deltaTime = reader.readFloat();
    model.viewWidth = reader.readFloat();
    model.viewHeight = reader.readFloat();
    model.touchStartX = reader.readFloat();
    model.touchStartY = reader.readFloat();
    model.lastTouchX = reader.readFloat();
    model.lastTouchY = reader.readFloat();
    model.touchX = reader.readFloat();
    model.touchY = reader.readFloat();
  }

  @override
  void write(BytesWriter writer, Test model) {
    writer.writeFloat(model.deltaTime);
    writer.writeFloat(model.viewWidth);
    writer.writeFloat(model.viewHeight);
    writer.writeFloat(model.touchStartX);
    writer.writeFloat(model.touchStartY);
    writer.writeFloat(model.lastTouchX);
    writer.writeFloat(model.lastTouchY);
    writer.writeFloat(model.touchX);
    writer.writeFloat(model.touchY);
  }

  @override
  void skip(BytesReader reader, int count) {
    for (int i = 0; i < count; i += 1) {
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
    }
  }
}

class TestIntoToBuffers implements IntoToBuffers<Test> {
  const TestIntoToBuffers()

  @override
  Test read(BytesReader reader) {
    final deltaTime = reader.readFloat();
    final viewWidth = reader.readFloat();
    final viewHeight = reader.readFloat();
    final touchStartX = reader.readFloat();
    final touchStartY = reader.readFloat();
    final lastTouchX = reader.readFloat();
    final lastTouchY = reader.readFloat();
    final touchX = reader.readFloat();
    final touchY = reader.readFloat();

    return Test(
      deltaTime: deltaTime;
      viewWidth: viewWidth;
      viewHeight: viewHeight;
      touchStartX: touchStartX;
      touchStartY: touchStartY;
      lastTouchX: lastTouchX;
      lastTouchY: lastTouchY;
      touchX: touchX;
      touchY: touchY;
    );
  }

  @override
  void write(BytesWriter writer, Test model) {
    writer.writeFloat(model.deltaTime);
    writer.writeFloat(model.viewWidth);
    writer.writeFloat(model.viewHeight);
    writer.writeFloat(model.touchStartX);
    writer.writeFloat(model.touchStartY);
    writer.writeFloat(model.lastTouchX);
    writer.writeFloat(model.lastTouchY);
    writer.writeFloat(model.touchX);
    writer.writeFloat(model.touchY);
  }

  @override
  void skip(BytesReader reader, int count) {
    for (int i = 0; i < count; i += 1) {
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
      reader.readFloat();
    }
  }
}
